spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -122,14 +124,22 @@ public Firestore firestore(FirestoreOptions firestoreOptions) {
 
 		@Bean
 		@ConditionalOnMissingBean
-		public FirestoreMappingContext firestoreMappingContext() {
-			return new FirestoreMappingContext();
+		public FirestoreMappingContext firestoreMappingContext(FirestoreClassMapper mapper) {
+			return new FirestoreMappingContext(mapper);
 		}
 
 		@Bean
 		@ConditionalOnMissingBean
-		public FirestoreTemplate firestoreTemplate(FirestoreGrpc.FirestoreStub firestoreStub) {
-			return new FirestoreTemplate(firestoreStub, GcpFirestoreAutoConfiguration.this.firestoreRootPath);
+		public FirestoreClassMapper firestoreClassMapper(FirestoreOptions firestoreOptions, Firestore firestore) {
+			return new FirestoreClassMapper(firestoreOptions, (FirestoreRpc) firestoreOptions.getRpc());
```

Comment:
Can/should the `FirestoreRpc` itself be a bean? When you call `options.getRpc()` does it actually create resources or only returns an existing object?